src/components/ui4/ButtonUi4.tsx

```diff
@@ -145,6 +149,8 @@ export function ButtonUi4(props: Props) {
     return retStyle
   }, [disabled, hideContent, layout, mini, styles, type])
 
+  const spinnerStyle = React.useMemo(() => [styles.spinner, customMarginPadding], [customMarginPadding, styles.spinner])
```

Comment:
This `useMemo` does nothing. You can read the source code for `ActivityIndicator` yourself:

https://github.com/facebook/react-native/blob/5761ad6437b29dc559fce0914f2a8e71d02d25b3/Libraries/Components/ActivityIndicator/ActivityIndicator.js#L111

As you can see, the styles go through a `StyleSheet.compose` call before they head down to native code. On top of that, the `ActivityIndicator` itself doesn't use `React.memo` or `React.PureComponent` in any way.

Since the `useMemo` call isn't free, fake "optimizations" like these actually slow things down. So, just do it the easy way and write `<Xyz style={[a, b, c]} />` when you need multiple styles in one. The simplest option is also the fastest.

Reply:
Ok I'll keep that in mind for the future. I'm pretty sure I've gotten comments of this nature from Paul before when trying to do simply `<Xyz style={[a, b, c]} />`.... Case specific? Either way I'll keep an eye out